### PR TITLE
#167 Add scroll to error to formik-elements

### LINKED ISSUE #167: Add scrollToError to formik fields
## Basic information

* Tiller version:
  1.7.2
* Module:
  Storybook - Formik

## Enhancement description

Add scroll to first error found when submitting form.

### Possible workarounds
Add
```
<Form>
  <ScrollToError />
    ....
```

example of ScrollToError

```
export function ScrollToError() {
  const formik = useFormikContext();
  const submitting = formik?.isSubmitting;
  const errors = formik?.errors;

  useEffect(() => {
    const firstErrorField = Object.keys(errors)[0];
    const el = document.getElementsByName(firstErrorField)[0];
    (el?.parentElement ?? el)?.scrollIntoView({ behavior: "smooth", block: "center" });
  }, [submitting]);
  return null;
}
```

### PR DESCRIPTION
**Pull request by fbeceic**
## Basic information

* Tiller version: 1.8.0
* Module: formik-elements

## Description
Added the `useScrollToError` hook to the `formik-elements` module for rendering inside an arbitrary form. 

Introduced the `scrollToError` prop to the `FormContainer` component, providing an option to enable automatic scrolling to the first error via a boolean value. This addition enhances modularity by offering an alternative method for enabling error scrolling without the need for explicit component imports and rendering.

### Related issue

Closes #167

## Types of changes

- Enhancement (non-breaking change which enhances existing functionality)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the project's **CONTRIBUTING** document
- [x] I have checked my code with the project's static analysis tooling
- [x] I have formatted my code with the project's Prettier code-style configuration
- [x] I have checked my code for misspellings
- [x] I have organized my changes in easy-to-follow commits
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added a story to cover my changes
- [x] All new and existing story tests pass

### CODE REVIEW

File: libs/formik-elements/src/ScrollToError.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+import { useEffect } from "react";
+import { useFormikContext } from "formik";
+
+export default function ScrollToError() {
```
Comment (jtomic-croz) [inline libs/formik-elements/src/ScrollToError.tsx:21]: make it a hook, not a component

File: libs/selectors/src/Select.tsx [hunk 2]
```diff
@@ -409,7 +409,7 @@ function Select<T>({
         >
           <div className={tokens.Button.container}>
             <div className={tokens.Button.value}>{valueLabel}</div>
-            <input className={tokens.Button.input} ref={inputRef} tabIndex={-1} />
+            <input name={name} className={tokens.Button.input} ref={inputRef} tabIndex={-1} />
```
Comment (jtomic-croz) [inline libs/selectors/src/Select.tsx:412]: why is this in this PR ? If you found another error, please create a separate issue and PR

File: libs/formik-elements/src/FormContainer.tsx [hunk 3]
```diff
@@ -20,9 +20,26 @@ import * as React from "react";
 import { Formik, FormikConfig, FormikHelpers } from "formik";
 
 import { createNamedContext, ValidationError } from "@tiller-ds/util";
+import ScrollToError from "./ScrollToError";
 
 type FormikProps<V = any> = {
+  /**
+   * Specifies whether validation should occur after form submission.
```
Comment (jtomic-croz) [inline libs/formik-elements/src/FormContainer.tsx:27]: "after form submission" or what? Isn't validation after form submission always happening?

File: libs/formik-elements/src/FormContainer.tsx [hunk 4]
```diff
@@ -20,9 +20,26 @@ import * as React from "react";
...
+   * Specifies whether the form should automatically scroll to the first validation error encountered.
+   *
+   * **Note**: Enabling this feature will cause the form to scroll to the first validation error found
+   * in your validation schema. It's important to ensure that the validationSchema prop defines
+   * validations in the same order as your form fields are rendered to ensure accurate scrolling.
+   *
+   * @type {boolean}
+   * @default false
+   */
+  scrollToError?: boolean;
```
Comment (jtomic-croz) [inline libs/formik-elements/src/FormContainer.tsx:42]: please add an adequate story to show off this behaviour

File: libs/formik-elements/src/FormContainer.tsx (hunk 3 above)
Comment (fbeceic) [reply libs/formik-elements/src/FormContainer.tsx:27]: No, default behavior is onBlur of a component.

Comment (jtomic-croz) [reply libs/formik-elements/src/FormContainer.tsx:27]: please add that to the prop description

**Review by jtomic-croz — APPROVED** [review]

**Pull request merged**